auth/jwt/README.md

```diff
@@ -0,0 +1,123 @@
+# package auth/jwt
+
+`package auth/jwt` provides a set of interfaces for service authorization through [JSON Web Tokens](https://jwt.io/).
+
+## Usage
+
+NewParser takes a key function and an expected signing method and returns an `endpoint.Middleware`. 
+The middleware will parse a token passed into the context via the `jwt.JWTTokenContextKey`. 
+If the token is valid, any claims will be added to the context via the `jwt.JWTClaimsContextKey`.
+
+```go
+import (
+	stdjwt "github.com/dgrijalva/jwt-go"
+    
+    "github.com/go-kit/kit/auth/jwt"
+    "github.com/go-kit/kit/endpoint"
+)
+
+func main() {
+	var exampleEndpoint endpoint.Endpoint
+	{
+		keyFunc := func(token *stdjwt.Token) (interface{}, error) { return []byte("SigningString"), nil }
+		jwtParser := jwt.NewParser(keyFunc, stdjwt.SigningMethodHS256)
+        
+		exampleEndpoint = MakeExampleEndpoint(service)
+		exampleEndpoint = jwtParser(exampleEndpoint)
+	}
+}
+```
+
+NewSigner takes a JWT key id header, the signing key, signing method, and a claims object. It returns an `endpoint.Middleware`.
+The middleware will build the token string and add it to the context via the `jwt.JWTTokenContextKey`.
+
+```go
+import (
+	stdjwt "github.com/dgrijalva/jwt-go"
+    
+    "github.com/go-kit/kit/auth/jwt"
+    "github.com/go-kit/kit/endpoint"
+)
+
+func main() {
+	var exampleEndpoint endpoint.Endpoint
+	{
+		jwtSigner := jwt.NewSigner("kid-header", []byte("SigningString"), stdjwt.SigningMethodHS256, jwt.Claims{})
+        
+		exampleEndpoint = grpctransport.NewClient(
+        	. // build client endpoint here
+			.
+			.
+		).Endpoint()
+
+		exampleEndpoint = jwtSigner(exampleEndpoint)
```

Comment:
Like above, what do you think about inlining the jwtSigner?

```go
exampleEndpoint = jwt.NewSigner(
    "kid-header",
    []byte("SigningString"),
    stdjwt.SigningMethodHS256,
    jwt.Claims{},
)(exampleEndpoint)
```

Maybe that's too long or worse than the alternative -- totally your call.